core/src/main/java/org/apache/iceberg/io/RollingDeleteWriter.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.io;
+
+import java.util.List;
+import org.apache.iceberg.DeleteFile;
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.StructLike;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.util.CharSequenceSet;
+
+abstract class RollingDeleteWriter<T, W extends FileWriter<T, DeleteWriteResult>>
```

Comment:
I comparing this to the `RollingDataWriter`, should we introduce a `DeleteWriter` (similar to `DataWriter`)?

Reply:
We currently have 2 delete writers: `EqualityDeleteWriter` and `PositionDeleteWriter`. I made them both extend the newly added `FileWriter` interface.

Whenever I needed an arbitrary delete writer, I used `FileWriter<T, DeleteWriteResult>`. We could add a separate interface called `DeleteWriter` but it will be empty and just parameterize `FileWriter`.

I don't have a preference here.